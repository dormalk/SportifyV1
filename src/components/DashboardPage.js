import React from 'react';
import { connect } from 'react-redux';
import {startLogout} from '../actions/auth';

export const DashboardPage = ({startLogout}) => (
    <div>
        <p>wellcome my frined</p>
        <button onClick={startLogout}>Logout</button>
    </div>
);


const mapPropsToDispatch = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});


export default connect(undefined,mapPropsToDispatch)(DashboardPage);
