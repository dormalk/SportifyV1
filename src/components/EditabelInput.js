import React from 'react';
//Test
export default class EditabelInput extends React.Component{
    constructor(props){
        super();
        this.state = {
            disable: true
        }
    }

    onEditClick = () => {
        const disable = !this.state.disable;
        this.setState({disable});
    }

    render(){
        return(
            <div>
                <label><span className="input_label">{this.props.label}</span>
                    <input type="text" value={this.props.val} onChange={this.props.onChangeValue} disabled={this.state.disable}/>
                    {this.state.disable? (
                        <span className="start__edit_input" onClick={this.onEditClick}>ערוך/י</span>
                    ):(
                        <span  className="end__edit_input" onClick={this.onEditClick}>סיימ/י עריכה</span>           
                    )}
                </label>
            </div>
        )        
    }
}
