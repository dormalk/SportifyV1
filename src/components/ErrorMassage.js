import React from 'react';


export default (props) => (
    <div>
    שגיאה התרחשה - {props.error}
    </div>
) 
