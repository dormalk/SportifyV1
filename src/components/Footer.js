import React from 'react';


export default () => (
    <div className="show-for-desktop footer">
        <span>כל הזכויות שמורות</span>
    </div>
)