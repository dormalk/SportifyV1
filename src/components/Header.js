import React from 'react';
import { connect } from 'react-redux';

export const Header = () => (
    <div className="header">
        <div className="content-container">
            <h1 className="header__title">Sportify</h1>
        </div>
    </div>
);

export default Header;
