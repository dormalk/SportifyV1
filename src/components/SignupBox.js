import React from 'react';
import { connect } from 'react-redux';
import { startUploadImage } from '../actions/storage';

export class SignupBox extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            user:{
                fname: this.props.fname || '',
                lname: this.props.lname || '',
                email: this.props.email || '',
                password: this.props.password || '',        
            },
            c_password: '',
            error: ''
        }
    }

    onSubmit = (e) => {
        e.preventDefault();
    };

    onFnameChange = (e) => {
        const fname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, fname } } ));
    }

    onLnameChange = (e) => {
        const lname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, lname } } ));
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, email } } ));
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        if(password.length < 20) {
            this.setState(() => ( { user: { ...this.state.user, password } } ));
        }
    }

    onConfirmPasswordChange = (e) => {
        const c_password = e.target.value;
        if(c_password.length < 20) {
            this.setState(() => ( { c_password } ));
        }
    }

    checkCompatibilityPasswords = (e) => {
        const r_password = this.state.user.password;
        const c_password = this.state.c_password;
        if(c_password !== r_password){
            this.setState(() => ({ error:'Passwords do not match' }));
        }
        else{
            this.setState(() => ({ error:'' }));
        }
    }

    onChangeImagePicker = (e) => {
        e.preventDeafule();
        const picture = e.target.files[0];
        if(picture.type !== 'image/jpeg' && picture.type !== 'image/jpg' && picture.type !== 'image/png'){
            this.setState({error:'נא בחר קובץ תמונה (jpg,jpeg,png)'});
        }
        else if(picture.size > 585750){
            this.setState({error:'נא בחר קובץ תמונה פחות מ 585,750 בית'});            
        }
        else{
            this.setState(() => ({error:''}));  
            this.props.startUploadImage(picture);        
        }
    }

    render(){
        return (
            <div className="content-container">
                <form 
                    onSubmit={this.onSubmit}
                    className="signup-form"
                >
                    <h2>CREATE NEW ACCOUNT</h2>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <label>First Name:
                        <input 
                            type="text"
                            onChange={this.onFnameChange}
                            required
                        />
                    </label>
                    <label>Last Name:
                        <input 
                            type="text"
                            onChange={this.onLnameChange}
                            required
                        />
                    </label>
                    <label>Email:
                        <input 
                            type="text"
                            onChange={this.onEmailChange}
                            required
                        />
                    </label>
                    <label>Password:
                        <input 
                            type="password"
                            onChange={this.onPasswordChange}
                            required
                        />
                    </label>
                    <label>Confirm Password:
                        <input 
                            type="password"
                            onChange={this.onConfirmPasswordChange}
                            onBlur={this.checkCompatibilityPasswords}
                            required
                        />
                    </label>
                    <button className="signup__box">CREATE ACCOUNT</button>
                </form>
            </div>
        );
    };
};

const mapPropsToDispatch = (dispatch) => {
    startUploadImage: (picture) => dispatch(startUploadImage(picture));
}


export default connect(undefined,mapPropsToDispatch)(SignupBox);