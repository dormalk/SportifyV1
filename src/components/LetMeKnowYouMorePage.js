//Test
import React from 'react';
import { connect } from 'react-redux';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import {isPartOfHobbie,isHobbie,fields} from '../selectors/HobbiesSuggestion'; 
import EditabelInput from './EditabelInput';


export const CheckDetail = ({fname,lname,onChangeFname,onChangeLname,onContinueButton}) => (
    <div className="check_detail">
        <h2>האם הפרטים נכונים?</h2>
        <EditabelInput val={fname} onChangeValue={onChangeFname} label={"שם פרטי"}/>
        <EditabelInput val={lname} onChangeValue={onChangeLname} label={"שם משפחה"}/>
        <button className="confirm_detail" onClick={onContinueButton}>אישור</button>
    </div>
)

export const YourGender = ({hendelGenderChange}) => {
    const genders = ['זכר','נקבה','לא מעוניין למסור']
    return(
        <div className="your_gender">
            <h2>מהו מינך?</h2>
            {genders.map((gender,i) => (
                <label key={i}>
                    <input 
                        key={i}
                        onChange={hendelGenderChange}
                        type='radio'
                        name='gender'
                        value={gender}/>
                        <span className="radiomark"><span className="radiomark_checked"></span></span>
                        {gender}
                </label>
            ))}
    </div>
    )
}

export const YourAge = ({hendelAgeChange}) => (
    <div>
        <h2>גיל</h2>
        <input type="number" placeholder="18" min="16" max="99" onChange={hendelAgeChange}/>
    </div>
);

export const YourHobbies = ({currHobbie,Hobbies,onHobbieEntered,onHobbieRemoved,onHobbieChanged,onHobbieChangedFromSuggest,suggestHobbie}) => (
    <div>
        <h2>מה התחביבים שלך?</h2>
        <input type="text" value={currHobbie} onChange={onHobbieChanged.bind(this)} onKeyPress={onHobbieEntered.bind(this)}/>
        {Hobbies.map((hobbie,i) => (<div key={i}>{hobbie}<button onClick={onHobbieRemoved.bind(this,hobbie)}>x</button></div>))}
        {currHobbie!='' && suggestHobbie(currHobbie).map((val,i) => (<div key={i} onClick={onHobbieChangedFromSuggest.bind(this,val)}>{val}</div>))}
    </div>
)

export class LetMeKnowYouMorePage extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            user: {
                ...this.props.user,
                detail: {
                    gender: '',
                    age:'',
                    hobbies: []    
                }
            },
            autocompleat:'',
            flow: 1,
        }
        
        this.isStateHobbie = this.isStateHobbie.bind(this);
    }
    
    onChangeFname = (e) => {
        const fname = e.target.value;
        const user = {
            ...this.state.user,
            fname,
            detail: {
               ...this.state.user.detail 
            }
        }
        this.setState({user});
    };


    onChangeLname = (e) => {
        const lname = e.target.value;
        const user = {
            ...this.state.user,
            lname,
            detail: {
               ...this.state.user.detail 
            }
        }
        this.setState({user});
    };


    hendelGenderChange = (e) => {
        const gender = e.target.value;
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               gender 
            }
        }
        this.setState({user});
    }

    hendelAgeChange = (e) => {
        const age = e.target.value;
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               age
            }
        }
        this.setState({user});
    }

    isStateHobbie = (value) => {
        var _isHobbie = true;
        const _fields = this.state.user.detail.hobbies;
        for(var i=0;i<_fields.length;i++)
            if(_fields[i] === value){
                _isHobbie = false;
                break;
            } 
        return _isHobbie;    
    }

    onHobbieEntered = (e) => {
        const hobbie = e.target.value;
        if(e.key === 'Enter'){
            if(isHobbie(hobbie) && this.isStateHobbie(hobbie)){
                const autocompleat = '';
                const hobbies = this.state.user.detail.hobbies;
                hobbies.push(hobbie);    
                const user = {
                    ...this.state.user,
                    detail: {
                       ...this.state.user.detail,
                       hobbies
                    }
                }        
                this.setState({user});
                this.setState({autocompleat})    
            }
        }
    }

    onHobbieRemoved = (hobbie) => {
        var hobbies = this.state.user.detail.hobbies.filter((h) => {return h !== hobbie;});
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               hobbies
            }
        }        
        this.setState({user});
    }

    onHobbieChanged = (e) => {
        const autocompleat = e.target.value;
        if(isPartOfHobbie(autocompleat))
            this.setState({autocompleat});
    }

    suggestHobbie = (value) => {
        return fields.filter((v) => {return v.startsWith(value) && this.isStateHobbie(v)});
    }

    onHobbieChangedFromSuggest = (value) => {
        const hobbie = value;
        const hobbies = this.state.user.detail.hobbies;
        hobbies.push(hobbie);
        const autocompleat = '';
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               hobbies
            }
        }        
        this.setState({user});
        this.setState({autocompleat});
    }


    onContinueButton = () => {
        this.setState({flow: this.state.flow+1});
    }

    render(){
        return this.state.user &&
        (
            <div className="content-container">
                <div className="let_me_know_you_more">
                    {this.state.user.fname && <h1>היי  {this.state.user.fname}, אני רוצה להכיר אותך יותר!</h1>}
                    <ReactCSSTransitionGroup
                        transitionName="exm"
                        transitionAppear={false}
                        transitionEnterTimeout={500}
                        transitionEnter={true}
                        transitionLeave={false}>
                        {
                            this.state.flow === 1 && 
                            <div>
                                <CheckDetail 
                                    fname={this.state.user.fname} 
                                    lname={this.state.user.lname}
                                    onChangeFname={this.onChangeFname}
                                    onChangeLname={this.onChangeLname}
                                    onContinueButton={this.onContinueButton}/>
                            </div>
                        }
                        {this.state.flow === 2 && <div className="grey_border_box_display">
                            <ReactCSSTransitionGroup
                            transitionName="exm"
                            transitionAppear={false}
                            transitionEnterTimeout={500}
                            transitionEnter={true}
                            transitionLeave={false}>
                                <YourGender hendelGenderChange={this.hendelGenderChange}/>
                                {this.state.user.detail.gender && <YourAge hendelAgeChange={this.hendelAgeChange}/>}
                                {this.state.user.detail.age && <YourHobbies 
                                                        Hobbies={this.state.user.detail.hobbies} 
                                                        onHobbieEntered={this.onHobbieEntered}
                                                        onHobbieRemoved={this.onHobbieRemoved}
                                                        onHobbieChanged={this.onHobbieChanged}
                                                        currHobbie={this.state.autocompleat}
                                                        onHobbieChangedFromSuggest={this.onHobbieChangedFromSuggest}
                                                        suggestHobbie={this.suggestHobbie}/>}
                                {this.state.user.detail.hobbies.length > 0 &&
                                <button>זהו סיימתי</button>}
                            </ReactCSSTransitionGroup>
                        </div>}
                    </ReactCSSTransitionGroup>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    user: state.user
});

export default connect(mapStateToProps,undefined)(LetMeKnowYouMorePage);
