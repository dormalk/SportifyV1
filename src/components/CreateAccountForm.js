import React from 'react';
import { connect } from 'react-redux';
import { startUploadImage } from '../actions/storage';
import { startCreateNewUser } from '../actions/user';
import ErrorMassage from './ErrorMassage';


export class CreateAccountForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            user:{
                fname: this.props.fname || '',
                lname: this.props.lname || '',
                email: this.props.email || '',
                password: this.props.password || '',
                profile: this.props.profile || ''       
            },
            c_password: '',
            error: ''
        }
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.startCreateNewUser(this.state.user).then(() => {
            this.setState(() => ({ error:'' }));
        }).catch((error) => {
            if(error.code === 'auth/email-already-in-use'){
                this.setState(() => ({ error:'המייל שהזנת נימצא בשימוש' }));
            }
            
        });
    };

    onFnameChange = (e) => {
        const fname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, fname } } ));
    }

    onLnameChange = (e) => {
        const lname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, lname } } ));
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, email } } ));
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        if(password.length < 20) {
            this.setState(() => ( { user: { ...this.state.user, password } } ));
        }
    }

    onConfirmPasswordChange = (e) => {
        const c_password = e.target.value;
        if(c_password.length < 20) {
            this.setState(() => ( { c_password } ));
        }
    }

    checkCompatibilityPasswords = (e) => {
        const r_password = this.state.user.password;
        const c_password = this.state.c_password;
        if(c_password !== r_password){
            this.setState(() => ({ error:'Passwords do not match' }));
        }
        else{
            this.setState(() => ({ error:'' }));
        }
    }

    onProfileChange = (e) => {
        const picture = e.target.files[0];
        if(picture.type !== 'image/jpeg' && picture.type !== 'image/jpg' && picture.type !== 'image/png'){
            this.setState({error:'נא בחר קובץ תמונה (jpg,jpeg,png)'});
        }
        else if(picture.size > 585750){
            this.setState({error:'נא בחר קובץ תמונה פחות מ 585,750 בית'});
            console.log('Hello');
        }
        else{
                this.setState(() => ({error:''}));  
                this.props.startUploadImage(picture).then((profile) => {
                    this.setState({error:''});
                    this.setState(() => ( { user: { ...this.state.user, profile } } ));
                }).catch((error) => {
                    this.setState({error:'לא ניתן לעלות קובץ תמונה ברגע זה'});
                });
        }
    }

    render(){
        return (
            <div>
                {this.state.error && <ErrorMassage error={this.state.error}/>}
                <div className="content-container">
                    <div className="signup-box">
                        <form 
                            onSubmit={this.onSubmit}
                            className="signup-form"
                        >
                            <h2>יצירת משתמש חדש</h2>
                            {this.state.error && <p className="form__error">{this.state.error}</p>}
                            <label>שם פרטי
                                <input 
                                    type="text"
                                    onChange={this.onFnameChange}
                                    required
                                />
                            </label>
                            <label>שם משפחה
                                <input 
                                    type="text"
                                    onChange={this.onLnameChange}
                                    required
                                />
                            </label>
                            <label>אימייל
                                <input 
                                    type="text"
                                    onChange={this.onEmailChange}
                                    required
                                />
                            </label>
                            <label>סיסמא
                                <input 
                                    type="password"
                                    onChange={this.onPasswordChange}
                                    required
                                />
                            </label>
                            <label>אימות סיסמא
                                <input 
                                    type="password"
                                    onChange={this.onConfirmPasswordChange}
                                    onBlur={this.checkCompatibilityPasswords}
                                    required
                                />
                            </label>
                            <label>תמונת פרופיל
                                <input type="file" onChange={this.onProfileChange}/>
                            </label>
                            {this.state.profile && <img src={this.state.profile}/>}
                            <button className="signup__box">צור חשבון</button>
                        </form>
                    </div>
                </div>
            </div>
        );
    };
};

const mapPropsToDispatch = (dispatch) => ({
    startUploadImage: (picture) => dispatch(startUploadImage(picture)),
    startCreateNewUser: (user) => dispatch(startCreateNewUser(user))
});


export default connect(undefined,mapPropsToDispatch)(CreateAccountForm);