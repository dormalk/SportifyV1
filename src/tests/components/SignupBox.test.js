import React from 'react';
import { shallow } from 'enzyme';
import { SignupBox } from '../../components/SignupBox';

test('should render SignInBox correctly', () => {
    const wrapper = shallow(<SignupBox/>);
    expect(wrapper).toMatchSnapshot();
});


test('should set first name on input change', () => {
    const value = 'israel';
    const wrapper = shallow(<SignupBox />);
    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('user').fname).toBe(value);
});

test('should set last name on input change', () => {
    const value = 'israeli';
    const wrapper = shallow(<SignupBox />);
    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('user').lname).toBe(value);
});


test('should set email on input change', () => {
    const value = 'example@gmail.com';
    const wrapper = shallow(<SignupBox />);
    wrapper.find('input').at(2).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('user').email).toBe(value);
});

test('should set password on input change', () => {
    const value = '1234567';
    const wrapper = shallow(<SignupBox />);
    wrapper.find('input').at(3).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('user').password).toBe(value);
});

test('should set confirmed password on input change', () => {
    const value = '1234567';
    const wrapper = shallow(<SignupBox />);
    wrapper.find('input').at(4).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('c_password')).toBe(value);
});


test('should set error if passwords not match', () => {
    const password1 = '1234567';
    const password2 = 'q1w2e3';
    const wrapper = shallow(<SignupBox />);
    wrapper.find('input').at(3).simulate('change', {
        target: { value:password1 }
    });

    wrapper.find('input').at(4).simulate('change', {
        target: { value:password2 }
    });

    expect(wrapper.state('erroe')).not.toBe('');
});