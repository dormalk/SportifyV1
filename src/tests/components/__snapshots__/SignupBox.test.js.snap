// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignInBox correctly 1`] = `
<div
  className="content-container"
>
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <h2>
      CREATE NEW ACCOUNT
    </h2>
    <label>
      First Name:
      <input
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Last Name:
      <input
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Email:
      <input
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Password:
      <input
        onChange={[Function]}
        required={true}
        type="password"
      />
    </label>
    <label>
      Confirm Password:
      <input
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="password"
      />
    </label>
    <button
      className="signup__box"
    >
      CREATE ACCOUNT
    </button>
  </form>
</div>
`;
